floodlight/filter/filter.py

```diff
@@ -0,0 +1,223 @@
+from typing import Dict
+from floodlight.utils.types import Numeric
+
+import numpy as np
+import scipy.signal
+
+from floodlight import XY
+
+
+def _get_sequences_without_nans(data_column: np.ndarray) -> np.ndarray:
+    """Gets indices of sequences without NaNs from a sequence of data.
+
+    Parameters
+    ----------
+    data_column: np.ndarray
+        One-dimensional array of data with alternating sequences of Numbers and NaNs
+
+    Returns
+    -------
+    sequences: np.ndarray
+        Two-dimensional array with nx2 entrys of n sequences without NaNs and n[0] being
+        the first frame of the sequences and n[1] being the last frame (excluded) of the
+        sequence.
+    """
+
+    # check if even or odd sequences contain NaNs
+    if np.isnan(data_column[0]):
+        first_sequence = 1
+    else:
+        first_sequence = 0
+
+    # indices where nans and numbers are next to each other
+    change_points = np.where(
+        np.diff(np.isnan(data_column), prepend=np.nan, append=np.nan)
+    )[0]
+    # sequences without nans
+    sequences = np.array(
+        [
+            (change_points[i], change_points[i + 1])
+            for i in range(len(change_points) - 1)
+        ]
+    )[first_sequence::2]
+
+    return sequences
+
+
+def _filter_sequence_butterworth_lowpass(
+    signal: np.ndarray, order: int = 3, cutoff: Numeric = 1, framerate: Numeric = None
+) -> np.ndarray:
+    """Filters a signal with a analog Butterworth low pass filter.
+
+    Parameters
+    ----------
+    signal: np.ndarray
+        Sequence of coordinates in a xy-Object.
+    order: int
+        The order of the filter.
+    cutoff: Numeric
+        The critical frequency. If ``framerate`` is not specified, the cutoff is
+        normalized from 0 to 1, where 1 is the Nyquist frequency.
+    framerate: Numeric, optional
```

Comment:
what is it good for if I supply this?

Reply:
Also not described any further by scipy. I may refactor the documentation before the final version to indicated that this is just a wrapper for the scipy filter functions and refer to their docu